numpy/core/src/multiarray/descriptor.c

```diff
@@ -1215,14 +1209,16 @@ _convert_from_dict(PyObject *obj, int align)
     }
     new->elsize = totalsize;
     if (!PyTuple_Check(names)) {
-        names = PySequence_Tuple(names);
-    }
-    else {
-        Py_INCREF(names);
+        PyObject *tmp = PySequence_Tuple(names);
+        Py_DECREF(names);
+        names = tmp;
```

Comment:
What is the name of the macro that does this?

Reply:
PySETREF

Reply:
thanks